cybersecurity/log-intrusion-detector/devashmit/detector.py

```
@@ -0,0 +1,79 @@
+import argparse, re
+from collections import Counter, defaultdict
+from datetime import datetime
+from pathlib import Path
+
+BRUTE_FORCE_THRESHOLD = 5
+SCAN_404_THRESHOLD    = 10
+RAPID_REQ_THRESHOLD   = 30
+
+def parse_access_log(line):
+    m = re.match(r'(\S+) \S+ \S+ \[([^\]]+)\] "(\S+) (\S+) \S+" (\d+) \d+', line)
+    if not m: return None
```

Ruff (0.15.15)

[error] Multiple statements on one line (colon)

(E701)